### PR TITLE
issue-72 | HOC plop generator file updated

### PR DESCRIPTION
**Pull request by bhardwajamit0890**

### CODE REVIEW

File: generators/component/index.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ module.exports = {
       name: 'type',
       message: 'Select the type of component',
       default: 'Stateless Function',
-      choices: () => ['Stateless Function', 'React.Component'],
+      choices: () => ['Stateless Function', 'React.Component', 'hoc'],
```
Comment (vinodloha) [inline generators/component/index.js]: Let's keep the option as HOC

**Review by vinodloha — CHANGES_REQUESTED** [review]
Do this small change please

File: generators/component/hoc.js.hbs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// @flow
+/**
+ *
+ * {{ properCase name }}
+ *
+ */
+
+const {{ properCase name}} = Component => {
```
Comment (vinodloha) [inline generators/component/hoc.js.hbs]: This template is not enough. Have minimum imports. I.e React, types definitions etc. 
Rename Component as WrappedComponent.

Template should looks like below: 
```
import React from 'react';

import getDisplayName from '../../core/utils/getDisplayName';

const {{ properCase name}} = WrappedComponent => {
  // ...and returns another component...
  class extends React.Component {
    constructor(props) {
      super(props);
      this.state = {};
    }

    render() {
    
      return <WrappedComponent {...this.props} />;
    }
  };
  {{ properCase name}}.displayName = `{{ properCase name}}(${getDisplayName(WrappedComponent)})`;
  return {{ properCase name}};
}
``` 

Also create a file at core/utils with following content. 
function getDisplayName(WrappedComponent) {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
}

File: generators/utils/componentExists.js [hunk 3]
```diff
@@ -11,9 +11,11 @@ const atomComponents = fs.readdirSync(path.join(__dirname, '../../lib/components
 const moleculeComponents = fs.readdirSync(path.join(__dirname, '../../lib/components/molecules'));
 const organismComponents = fs.readdirSync(path.join(__dirname, '../../lib/components/organisms'));
 const templateComponents = fs.readdirSync(path.join(__dirname, '../../lib/components/templates'));
+const hoc = fs.readdirSync(path.join(__dirname, '../../lib/components/hoc'));
```
Comment (vinodloha) [inline generators/utils/componentExists.js:14]: I believe we are also creating test/* dir and files too.

**Review by vinodloha — CHANGES_REQUESTED** [review]

**Comment by bhardwajamit0890** [comment]
Done

**Pull request merged**